splice/web/api/content.py

```
@@ -0,0 +1,98 @@
+        return jsonify(message="Invalid content uploaded, expecting a zip file."), 400
+
+    if not name:
+        return jsonify(message="Missing content name in the query string."), 400
+
+    # try to fetch the content from the database
+    new_content = False
+    freeze = False
+    content_record = get_content(name)
+    if content_record is None:
+        content_record = {"name": name}
+        new_content = True
+
+    # determin the version based on the query string (if specified), and the current version in the database
+    if version:
+        try:
+            version = int(version)
+        except:
+            return jsonify(message="Invalid version number %s." % version), 400
+        else:
+            # make sure the version from the query string is sane
+            if version < 0:
+                return jsonify(message="Negative version number in the query string."), 400
+            if not new_content and content_record["version"] < version:
+                return jsonify(message="Given version %d is ahead of the controlled version %d." % (version, content_record["version"])), 400
+            elif new_content and version != 0:
+                return jsonify(message="Failed to find the content with the given version %d. You can omit the version to create a new content." % version), 400
+
+            if not new_content and version <= content_record["version"]:
+                freeze = True  # re-sign an existing version, do not bump up the version
+    else:
+        if new_content:
+            version = 0
+            content_record['version'] = version
+        else:
+            version = content_record['version']
+
+    # sign the content then upload it to S3
+    try:
+        # TODO(najiang@mozilla.com): return the signing pub key and store it to database?
+        urls, new_version = upload_signed_content(content_file.stream, name, version, freeze)
+    except Exception as e:
+        return jsonify(message="%s" % e), 400
```

The error may also be a 50X family, e.g. if an S3 upload failed.
It would be good to catch those. Also:

`except Exception as e` does not catch all exceptions in python 2.7. not all exceptions inherit from `Exception`.